include/revng/BasicAnalyses/GeneratedCodeBasicInfo.h

```diff
@@ -414,3 +443,31 @@ struct BlackListTrait<const GeneratedCodeBasicInfo &, llvm::BasicBlock *>
     return !this->Obj.isTranslated(Value);
   }
 };
+
+/// Analysis to compute GCBI.
+class GeneratedCodeBasicInfoAnalysis
+  : public llvm::AnalysisInfoMixin<GeneratedCodeBasicInfoAnalysis> {
+  friend llvm::AnalysisInfoMixin<GeneratedCodeBasicInfoAnalysis>;
+  static llvm::AnalysisKey Key;
+
+public:
+  using Result = GeneratedCodeBasicInfo;
+  Result run(llvm::Module &M, llvm::ModuleAnalysisManager &);
+  Result run(llvm::Function &F, llvm::FunctionAnalysisManager &);
+};
+
+/// Legacy pass manager pass to access GCBI.
+class GeneratedCodeBasicInfoWrapperPass : public llvm::ModulePass {
+  std::unique_ptr<GeneratedCodeBasicInfo> GCBI;
+public:
+  static char ID;
+
+  GeneratedCodeBasicInfoWrapperPass()
+    : llvm::ModulePass(ID) { }
+
+  GeneratedCodeBasicInfo &getGCBI() { return *GCBI; }
+  const GeneratedCodeBasicInfo &getGCBI() const { return *GCBI; }
+
+  bool runOnModule(llvm::Module &M) override;
+  void getAnalysisUsage(llvm::AnalysisUsage &AU) const override;
```

Comment:
I like to keep `getAnalysisUsage` body in the header so it "documents" the pass dependencies, if any.